docs/environment-variables.md

```diff
@@ -135,6 +135,10 @@ those.
   in parallel and deploy to specific nodes; each ID must be unique among the set
   of nodes. A single node should have the same value between consecutive restarts.
   Subgraphs get assigned to node IDs and are not reassigned to other nodes automatically.
+- `GRAPH_NODE_ENABLE_NEW_BEHAVIOR`: enables backwards-compatibility breaking
+  behavioural changes within Graph Node. Currently this only removes the node
+  id substitution behaviour. When set, this will disable transforming all
+  hyphens (-) to underscores (_).
```

Comment:
This should be called something more specific, say `GRAPH_NODE_LITERAL_NAME`; once this is out there, we can't add to the kind of things that this changes since we'll have a mix of people with this on and off who may or may not want to also use any behavior we would add.

The docs should say something like
```
(Docker only) Use the literal `node_id` provided to the docker start script instead of replacing hyphens (-) in names with underscores (_) Changing this for an existing `graph-node` installation requires also changing it in the `subgraphs.subgraph_deployment_assignment` table in the database.
```

It's a little weird in that this variable only has an effect with our `Dockerfile` where everything else is built into the `graph-node` binary but I don't have a better idea where to document this.

Reply:
My apologies for the delayed response. Family life.

With regards to the variable name, I agree, a more succinct variable name is called for. I propose `GRAPH_NODE_ID_USE_LITERAL_VALUE`. It's a bit longer, but a bit more clear. Ideally the effect of this variable could be removed as people (hopefully) pick up and adopt the changes.

And the behaviour isn't even in the Dockerfile, it's in that little bootstrapping script in the same directory. Its function would better be served by an entrypoint that does all the environmental stuff.